apps/meteor/client/hooks/useNotifyUser.ts

```diff
@@ -0,0 +1,92 @@
+import type { AtLeast, ISubscription } from '@rocket.chat/core-typings';
+import { useRouter, useStream, useUser, useUserPreference } from '@rocket.chat/ui-contexts';
+import { useCallback, useEffect } from 'react';
+
+import { CachedChatSubscription } from '../../app/models/client';
+import { KonchatNotification } from '../../app/ui/client/lib/KonchatNotification';
+import { RoomManager } from '../lib/RoomManager';
+import { fireGlobalEvent } from '../lib/utils/fireGlobalEvent';
+import { isLayoutEmbedded } from '../lib/utils/isLayoutEmbedded';
+
+export const useNotifyUser = () => {
+	const user = useUser();
+	const router = useRouter();
+	const notifyUserStream = useStream('notify-user');
+	const muteFocusedConversations = useUserPreference('muteFocusedConversations');
+
+	const notifyNewRoom = useCallback(
+		async (sub: AtLeast<ISubscription, 'rid'>): Promise<void> => {
+			if (!user || user.status === 'busy') {
+				return;
+			}
+
+			if ((!router.getRouteParameters().name || router.getRouteParameters().name !== sub.name) && !sub.ls && sub.alert === true) {
+				KonchatNotification.newRoom(sub.rid);
+			}
+		},
+		[router, user],
+	);
+
+	const notifyNewMessageAudio = useCallback(
+		(rid?: string) => {
+			const hasFocus = document.hasFocus();
+			const messageIsInOpenedRoom = RoomManager.opened === rid;
+
+			if (isLayoutEmbedded()) {
+				if (!hasFocus && messageIsInOpenedRoom) {
+					// Play a notification sound
+					void KonchatNotification.newMessage(rid);
+				}
+			} else if (!hasFocus || !messageIsInOpenedRoom || !muteFocusedConversations) {
+				// Play a notification sound
+				void KonchatNotification.newMessage(rid);
+			}
+		},
+		[muteFocusedConversations],
+	);
+
+	useEffect(() => {
+		if (!user?._id) {
+			return;
+		}
+
+		notifyUserStream(`${user?._id}/notification`, (notification) => {
+			const openedRoomId = ['channel', 'group', 'direct'].includes(router.getRouteName() || '') ? RoomManager.opened : undefined;
+
+			const hasFocus = document.hasFocus();
+			const messageIsInOpenedRoom = openedRoomId === notification.payload.rid;
+
+			fireGlobalEvent('notification', {
+				notification,
+				fromOpenedRoom: messageIsInOpenedRoom,
+				hasFocus,
+			});
+
+			if (router.getSearchParameters().layout === 'embedded') {
```

Comment:
should we use our util function from `client/lib` ?

Reply:
this router is from useRouter hook 🤔

Reply:
I don't get it. I wanted to mean, above this function we're using the [util function](https://github.com/RocketChat/Rocket.Chat/pull/33981/files/5ac85c030d1fa7ebc1fc810995e5b556b5f1544c#diff-1f3fe2aee19925b43ea380028dfaeb093aad014738ba744f820c30fb8df14873R35), why can't we use it again?

Reply:
ouch, I passed by line 35 without noticed... The router used in this lib is a deprecated way of usage